frontend/request.go

```diff
@@ -116,3 +119,74 @@ func marshalDockerfile(ctx context.Context, dt []byte, opts ...llb.ConstraintsOp
 	st := llb.Scratch().File(llb.Mkfile(dockerui.DefaultDockerfileName, 0600, dt), opts...)
 	return st.Marshal(ctx)
 }
+
+func ForwardToSigner(ctx context.Context, client gwclient.Client, platform *ocispecs.Platform, cfg *dalec.Frontend, s llb.State, filePattern string) (llb.State, error) {
+	const (
+		sourceKey  = "source"
+		contextKey = "context"
+		inputKey   = "input"
+
+		gatewayFrontend = "gateway.v0"
+	)
+
+	opts := client.BuildOpts().Opts
+	id := identity.NewID()
+
+	req, err := newSolveRequest(toFrontend(cfg))
+	if err != nil {
+		return llb.Scratch(), err
+	}
+
+	for k, v := range opts {
+		if k == "source" || k == "cmdline" {
+			continue
+		}
+		req.FrontendOpt[k] = v
+	}
+
+	inputs, err := client.Inputs(ctx)
+	if err != nil {
+		return llb.Scratch(), err
+	}
+
+	m := make(map[string]*pb.Definition)
+
+	for k, st := range inputs {
+		def, err := st.Marshal(ctx)
+		if err != nil {
+			return llb.Scratch(), err
+		}
+		m[k] = def.ToPB()
+	}
+	req.FrontendInputs = m
+
+	stateDef, err := s.Marshal(ctx)
+	if err != nil {
+		return llb.Scratch(), err
+	}
+
+	req.FrontendOpt[contextKey] = compound(inputKey, id)
+	req.FrontendInputs[id] = stateDef.ToPB()
+	req.FrontendOpt["dalec.target"] = opts["dalec.target"]
+	req.FrontendOpt["find.pattern"] = filePattern
```

Comment:
Same here, this should be in the spec as part of the arguments to the frontend.

Reply:
If we pass `dalec.target`, then the signer should know how to deal with the input implicitly and we should not need to pass a pattern to it.

Reply:
done.